src/vm/instruction.test.ts

```diff
@@ -0,0 +1,48 @@
+import { test, expect, describe } from 'bun:test';
+import { UnsignedInteger } from 'primitives/uint';
+import {
+  ApUpdate,
+  decodeInstruction,
+  FpUpdate,
+  Instruction,
+  Op1Src,
+  Opcode,
+  PcUpdate,
+  RegisterFlag,
+  ResLogic,
+} from './instruction';
+import { SignedInteger16 } from 'primitives/int';
+
+describe('Instruction', () => {
+  describe('decodeInstruction', () => {
+    test('should correctly decode a valid cairo instruction', () => {
+      const shift = 16n;
+      const offDst = 0b010101n;
+      const offOp0 = 0b101010n << shift;
+      const offOp1 = 0b111111n << (2n * shift);
+
+      const flag = 0b0100010000100100n << (3n * shift);
+
+      const encodedInstruction = offDst | offOp0 | offOp1 | flag;
+      const encodedInstructionUint64 =
```

Comment:
didn't we say it was not so elegant to put the type of a variable in its name?

Reply:
I think you wouldn't do it in rust because you can shadow, but can you do the same in ts?